.github/workflows/reject-updates-using-merge-commit.yml

```diff
@@ -0,0 +1,62 @@
+# This workflow ensures that pull requests aren't updated using merge commits.
+# To update a pull request with the latest changes you should rebase your branch on the base branch.
+# For more information, please refer to our contributing guide's section on 'Updating a pull request':
+# https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#updating-a-pull-request
+#
+# It is allowed to have merge commits in a pull request.
+# An example is a pull request that builds upon the work of another pull request.
+
+name: Reject Merge Commits
+on: [pull_request]
+
+permissions: {}
+
+jobs:
+  check-merge-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          filter: tree:0
+
+      - name: Check for invalid merge commits
+        run: |
+          base="origin/${{ github.event.pull_request.base.ref }}"
+          head=${{ github.event.pull_request.head.sha }}
+
+          # Find the common ancestor between the base and the head
+          common_ancestor=$(git merge-base $base $head)
+
+          # Get the list of merge commits in the pull request
+          merge_commits=$(git rev-list --merges $common_ancestor..$head)
+
+          # Check each merge commit
+          for merge_commit in $merge_commits; do
+            # Get the parent commits of the merge commit
+            parents=$(git rev-list --parents -n 1 $merge_commit | cut -d' ' -f2-)
+
+            # Check if any parent commit exists in the base branch
+            for parent in $parents; do
```

Comment:
:question: Could you explain (ideally in a code comment) why this is necessary? Why is it not enough to check whether there's _any_ merge commit?

Reply:
It's documented in the workflow (lines 6-7):

>It is allowed to have merge commits in a pull request.
># An example is a pull request that builds upon the work of another pull request.

Reply:
Oh my bad, so much for a careful review :grimacing: 

I'd prefer to also forbid this, stacked PRs + rebasing works just fine too. But that's mostly a personal preference, feel free to merge as is.

Reply:
I see. Currently we only allow merging pull requests with a merge commit through GH, but it would be possible with local merging and pushing.

I'm personally alright with supporting regular merge commits, but I'm happy to keep the discussion out of scope for now.